File: 547-number-of-provinces/number-of-provinces.py
class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        
        visited = [0 for i in range(len(isConnected))]
        count = 0
        def dfs(i):
            if visited[i] == 1:
                return 
            visited[i] = 1

            for c in range(len(isConnected[i])):
                if isConnected[i][c] == 1:
                    dfs(c)


        for r in range(len(isConnected)):
            for c in range(len(isConnected[0])):
                if visited[r] == 0:
                    count += 1
                    dfs(r)
        
        return count
                

File: 547-number-of-provinces/number-of-provinces.cpp
class Solution {
public:
    void dfs(vector<vector<int>>& graph,vector<int>& visited, int vertex){
        if (visited[vertex] == 1){
            return;
        }
        int n = graph.size();
        visited[vertex] = 1;
        for(int neighbor = 0; neighbor < n; neighbor++){
            if (graph[vertex][neighbor] == 1){
                dfs(graph, visited, neighbor);
            }
        }
    }

    int findCircleNum(vector<vector<int>>& isConnected) {
        int n = isConnected.size();
        vector<int> visited(n);
        int ans = 0;
        for(int vertex = 0; vertex < n; vertex++){
            if(visited[vertex] == 0){
                ans++;
                dfs(isConnected, visited, vertex);
            }
        }
        return ans;
    }
};